src/main/kotlin/second_pass/SymbolResolverVisitor.kt

```diff
@@ -0,0 +1,103 @@
+package second_pass
+
+import com.github.javaparser.ast.Node
+import com.github.javaparser.ast.body.ClassOrInterfaceDeclaration
+import com.github.javaparser.ast.body.ConstructorDeclaration
+import com.github.javaparser.ast.body.MethodDeclaration
+import com.github.javaparser.ast.body.VariableDeclarator
+import com.github.javaparser.ast.expr.FieldAccessExpr
+import com.github.javaparser.ast.expr.MethodCallExpr
+import com.github.javaparser.ast.visitor.VoidVisitorAdapter
+import first_pass.node_parser.FieldDeclarationParser
+import second_pass.context.ClassContext
+import second_pass.context.MemberContext
+import second_pass.context.VisitorContext
+import hierarchy.HierarchyGraph
+import second_pass.resolver.SymbolResolver
+import second_pass.signature.MemberDeclarationSignature
+import second_pass.signature.MemberSignature
+import util.ResolverVisitorResource
+
+class SymbolResolverVisitor : VoidVisitorAdapter<ClassContext>() {
+
+    private lateinit var visitorContext: VisitorContext
+    private lateinit var hierarchyVisitor: HierarchyVisitor
+    private lateinit var resolver: SymbolResolver
+
+    private fun initializeDataStructures(hierarchyGraph: HierarchyGraph) {
+        visitorContext = VisitorContext()
+        hierarchyVisitor = HierarchyVisitor(visitorContext, hierarchyGraph)
```

Comment:
Is this a visitor? Visitors are known for their visit method. If this is not a visitor, than neither should the VisitorContext have the word Visitor in its name.

**KEY INSIGHT**
While on the subject, the VisitorContext class in this current design is pretty dangerous and bug-prone, as we rely on other classes (hierarchyVisitor, resolver) to change the state of this object as it is being passed around (usually as a side-effect of the function).

Since the hierarchyVisitor has a single method that is primarily a call to the static class, do we need it? Can we make the SymbolResolverVisitor focus on the visiting, while the SymbolResolver (which we might call CaDETGraphConnector?) focuses on the linking of the hierarchy, accessed fields, and invoked methods?

If we follow the same pattern for the first_pass, we will have some node_connector static classes that will actually be the FieldAccessParser, MethodCallParser, and HierarchyParser (that contains the function from the current ClassDeclaration parser).